modules/partition-replicator/src/main/java/org/apache/ignite/internal/partition/replicator/PartitionReplicaLifecycleManager.java

```diff
@@ -210,6 +212,8 @@ public class PartitionReplicaLifecycleManager extends
     /** Configuration of rebalance retries delay. */
     private final SystemDistributedConfigurationPropertyHolder<Integer> rebalanceRetryDelayConfiguration;
 
+    private final ConcurrentMap<ZonePartitionId, ZonePartitionRaftListener> zonePartitionRaftListeners = new ConcurrentHashMap<>();
```

Comment:
It's an encapsulation leek, we should have raft listeners inside replicas. Why do you need it?

Reply:
I agree that this is somewhat of a encapsulation leek, but I think that code like this is even worse:

```
var zonePartitionReplicaListener = (ZonePartitionReplicaListener) replicaFut.join().listener();

zonePartitionReplicaListener.addTableReplicaListener(tablePartitionId, tablePartitionReplicaListenerFactory);
```

In my case I would also have to add special methods to access these listeners. Moreover, the lifecycle of raft and replica listeners is closely related - they are always added and removed together, so I would even prefer to have both of them in this map, this will make much easier to manage them. WDYT?

Reply:
I'd rather add something like addTableProcessor method to the replica in order to substitute 

`((ZonePartitionReplicaListener) replicaFut.join().listener()).addTableReplicaListener(tablePartitionId, createListener);` with something like `replicaFut.join().addTableProcessor(tablePartitionId, createListener, tableRaftListener)` in `PartitionReplicaLifecycleManager#loadTableListenerToZoneReplica`

`addTableReplicaListener` should be removed in that case.

Reply:
Makes sense

Reply:
I really don't like it. I'll create a ticket to refactor this. For now let stick with option you've proposed.

Reply:
I agree